commit 4c358e2f83922323269ca38743ded3711ceaf9cf (refs/changes/31/1731/1)
Author: Alex Wilson <alex.wilson@joyent.com>
Date:   2017-03-31T11:35:04-07:00 (2 years, 6 months ago)
    
    CNS-199 cnapi can return non-null, non-string "last_heartbeat" values

diff --git a/lib/cn-filter.js b/lib/cn-filter.js
index 3791d14..88f4258 100644
--- a/lib/cn-filter.js
+++ b/lib/cn-filter.js
@@ -142,7 +142,7 @@ function cutServerObj(obj) {
 	 * Don't bother looking for a heartbeat from a server that isn't
 	 * set up yet. These are always "down".
 	 */
-	if (obj.setup === false || obj.last_heartbeat === null) {
+	if (obj.setup === false || typeof (obj.last_heartbeat) !== 'string') {
 		cutObj.down = true;
 		return (cutObj);
 	}
